src/equations/ideal_glm_mhd_multiion_2d.jl

```diff
@@ -0,0 +1,1113 @@
+    total_electron_charge = zero(eltype(u_ll))
+    for k in eachcomponent(equations)
+        rho_k = u_ll[3 + (k - 1) * 5 + 1]
+        charge_ratio_ll[k] = rho_k * charge_to_mass[k]
+        total_electron_charge += charge_ratio_ll[k]
+    end
+    charge_ratio_ll ./= total_electron_charge
+
+    # Compute auxiliary variables
+    v1_plus_ll, v2_plus_ll, v3_plus_ll, vk1_plus_ll, vk2_plus_ll, vk3_plus_ll = charge_averaged_velocities(u_ll,
+                                                                                                           equations)
+    v1_plus_rr, v2_plus_rr, v3_plus_rr, vk1_plus_rr, vk2_plus_rr, vk3_plus_rr = charge_averaged_velocities(u_rr,
+                                                                                                           equations)
+
+    f = zero(MVector{nvariables(equations), eltype(u_ll)})
+
+    if orientation == 1
+        # Entries of Godunov-Powell term for induction equation (multiply by 2 because the non-conservative flux is 
+        # multiplied by 0.5 whenever it's used in the Trixi code)
+        f[1] = 2.0f0 * v1_plus_ll * B1_avg
+        f[2] = 2.0f0 * v2_plus_ll * B1_avg
+        f[3] = 2.0f0 * v3_plus_ll * B1_avg
+
+        for k in eachcomponent(equations)
+            # Compute term Lorentz term
+            f2 = charge_ratio_ll[k] * (0.5f0 * mag_norm_avg - B1_avg * B1_avg + pe_mean)
+            f3 = charge_ratio_ll[k] * (-B1_avg * B2_avg)
+            f4 = charge_ratio_ll[k] * (-B1_avg * B3_avg)
+            f5 = vk1_plus_ll[k] * pe_mean
+
+            # Compute multi-ion term (vanishes for NCOMP==1)
+            vk1_minus_ll = v1_plus_ll - vk1_plus_ll[k]
+            vk2_minus_ll = v2_plus_ll - vk2_plus_ll[k]
+            vk3_minus_ll = v3_plus_ll - vk3_plus_ll[k]
+            vk1_minus_rr = v1_plus_rr - vk1_plus_rr[k]
+            vk2_minus_rr = v2_plus_rr - vk2_plus_rr[k]
+            vk3_minus_rr = v3_plus_rr - vk3_plus_rr[k]
+            vk1_minus_avg = 0.5f0 * (vk1_minus_ll + vk1_minus_rr)
+            vk2_minus_avg = 0.5f0 * (vk2_minus_ll + vk2_minus_rr)
+            vk3_minus_avg = 0.5f0 * (vk3_minus_ll + vk3_minus_rr)
+            f5 += (B2_ll * (vk1_minus_avg * B2_avg - vk2_minus_avg * B1_avg) +
+                   B3_ll * (vk1_minus_avg * B3_avg - vk3_minus_avg * B1_avg))
+
+            # Compute Godunov-Powell term
+            f2 += charge_ratio_ll[k] * B1_ll * B1_avg
+            f3 += charge_ratio_ll[k] * B2_ll * B1_avg
+            f4 += charge_ratio_ll[k] * B3_ll * B1_avg
+            f5 += (v1_plus_ll * B1_ll + v2_plus_ll * B2_ll + v3_plus_ll * B3_ll) *
+                  B1_avg
+
+            # Compute GLM term for the energy
+            f5 += v1_plus_ll * psi_ll * psi_avg
+
+            # Add to the flux vector (multiply by 2 because the non-conservative flux is 
+            # multiplied by 0.5 whenever it's used in the Trixi code)
+            set_component!(f, k, 0, 2.0f0 * f2, 2.0f0 * f3, 2.0f0 * f4, 2.0f0 * f5,
```

Comment:
```suggestion
            set_component!(f, k, 0, 2 * f2, 2 * f3, 2 * f4, 2 * f5,
```
For integer quantities, you may just use integers (applies also to other areas in this file)